src/Microsoft.AspNetCore.Razor.Language/DefaultRazorIntermediateNodeLoweringPhase.cs

```diff
@@ -31,7 +32,7 @@ protected override void ExecuteCore(RazorCodeDocument codeDocument)
             var document = new DocumentIntermediateNode();
             var builder = IntermediateNodeBuilder.Create(document);
 
-            document.Options = _optionsFeature.GetOptions();
+            document.Options = codeDocument.GetCodeGenerationOptions() ?? _optionsFeature.GetOptions();
```

Comment:
This phase can still be used with the 2.0 code path, so it needs to support the old way of creating the options.